kiva/ps.py

```diff
@@ -240,7 +240,9 @@ def device_draw_image(self, img, rect):
         )
         self.contents.write("%.3f %.3f translate\n" % (left, top))
         # Rely on PIL's EpsImagePlugin to do the hard work here.
-        pil_img.save(self.contents, "eps", eps=0)
+        fp = BytesIO()
+        pil_img.save(fp, "eps", eps=0)
+        self.contents.write(fp.getvalue().decode('utf8'))
```

Comment:
I'm not entirely sure why we are saving the image to a `BytesIO` and then decoding it + writing it to `self.contents` - instead of saving it directly to `self.contents`.

Is this just best practice or is this preventing any possible issues?

Reply:
`self.contents` is a `StringIO` and `Image.save` will fail if you pass it in because this isn't Python 2 where you can write bytes into it. So we create a temporary `BytesIO` and then write it to `self.contents` after converting it to unicode.

Reply:
Ohh damn. So this was just straight up broken earlier and we just didn't know. Thanks!

Reply:
That's right